cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -515,6 +517,36 @@ Collection<WorkflowTable.WorkflowRunRecord> retrieveSpacedRecords(WorkflowId wor
   List<ProgramHistory> getRuns(Collection<ProgramId> programs, ProgramRunStatus status,
                                long startTime, long endTime, int limitPerProgram);
 
+
+  /**
+   * Get application state.
+   *
+   * @param request a {@link AppState} object with primary keys {namespace, appName, stateKey} set.
+   * @return state of application
+   */
+  Optional<byte[]> getState(AppState request);
+
+  /**
+   * Save application state.
+   *
+   * @param request a {@link AppState} object with all the fields set.
+   */
+  void saveState(AppState request);
+
+  /**
+   * Delete application state.
+   *
+   * @param request a {@link AppState} object with primary keys {namespace, appName, stateKey} set.
+   */
+  void deleteState(AppState request);
+
+  /**
+   * Delete all states related to an application.
+   *
+   * @param request a {@link AppState} object with primary keys {namespace, appName} set.
+   */
+  void deleteAllStates(AppState request);
```

Comment:
shouldn't be re-using the same AppState object for this method, but instead take a NamespaceId and an application name as arguments.